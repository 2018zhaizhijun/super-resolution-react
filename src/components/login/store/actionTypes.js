export const CHANGE_IS_VISIBLE = 'login/CHANGE_IS_VISIBLE';
//export const CHANGE_REGISTER_STATE = 'login/CHANGE_REGISTER_STATE';
export const CHANGE_LOGIN_STATE = 'login/CHANGE_USER_LOGIN_STATE';
export const CHANGE_AVATAR = 'login/CHANGE_AVATAR';
export const CHANGE_PROFILE = 'login/CHANGE_PROFILE_INFO';
export const CHANGE_TOKEN = 'login/CHANGE_PROFILE_TOKEN';
