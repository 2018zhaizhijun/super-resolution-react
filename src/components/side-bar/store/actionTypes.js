export const CHANGE_IS_VISIBLE = 'sidebar/CHANGE_IS_VISIBLE';
