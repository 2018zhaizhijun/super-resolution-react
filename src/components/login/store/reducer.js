import { Map } from 'immutable';
import * as actionTypes from './actionTypes';

const defaultState = Map({
    isVisible: false,
    //registering: false,
    isLogin: false,
    profile: {
        avatarUrl: null,
        usrname: null,
        telephone: null,
        sex: 0,
    },
    token: null,
});

function reducer(state=defaultState, action) {
    switch (action.type) {
        case actionTypes.CHANGE_IS_VISIBLE:
            return state.set('isVisible', action.payload);
        // case actionTypes.CHANGE_REGISTER_STATE:
        //     return state.set('registering', action.registering);
        case actionTypes.CHANGE_LOGIN_STATE:
            return state.set('isLogin', action.payload);
        // case actionTypes.CHANGE_AVATAR:
        //     return state.set('avatarUrl', action.avatarUrl);
        case actionTypes.CHANGE_PROFILE:
            return state.set('profile', {...state.get('profile'), ...action.payload});
        case actionTypes.CHANGE_TOKEN:
            return state.set('token', action.payload);
        default:
            return state;
    }
}

export default reducer;