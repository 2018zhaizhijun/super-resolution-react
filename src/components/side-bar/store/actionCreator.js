import * as actionTypes from './actionTypes';
import { createAction } from '@reduxjs/toolkit';

export const changeIsVisible = createAction(actionTypes.CHANGE_IS_VISIBLE);
// export const changeIsVisible = (visibleState) => ({
//     type: actionTypes.CHANGE_IS_VISIBLE,
//     isVisible: visibleState
// });
