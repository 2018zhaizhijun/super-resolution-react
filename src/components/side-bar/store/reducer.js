import { Map } from 'immutable';
import * as actionTypes from './actionTypes';

const defaultState = Map({
    isVisible: false,
});

function reducer(state=defaultState, action) {
    switch (action.type) {
        case actionTypes.CHANGE_IS_VISIBLE:
            return state.set('isVisible', action.payload);
        default:
            return state;
    }
}

export default reducer;