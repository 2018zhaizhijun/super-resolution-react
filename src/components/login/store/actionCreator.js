import { phoneLogin, getAvatar, getProfile, logout, sendAvatar } from '@service/login';
import * as actionTypes from './actionTypes';
import { message } from 'antd';
import { createAction } from '@reduxjs/toolkit';

import { sendUsrinfo } from '@service/login';

// 用createAction创建action可以不用再对其额外写单元测试
export const changeIsVisible = createAction(actionTypes.CHANGE_IS_VISIBLE);
// export const changeIsVisible = (visibleState) => ({
//     type: actionTypes.CHANGE_IS_VISIBLE,
//     isVisible: visibleState
// });

// export const changeRegisterState = (registerState) => ({
//     type: actionTypes.CHANGE_IS_VISIBLE,
//     registering: registerState
// });

export const changeLoginState = createAction(actionTypes.CHANGE_LOGIN_STATE);
// export const changeLoginState = (loginState) => ({
//     type: actionTypes.CHANGE_LOGIN_STATE,
//     isLogin: loginState
// });

// export const changeAvatar = (avatarUrl) => ({
//     type: actionTypes.CHANGE_AVATAR,
//     avatarUrl
// });

export const changeProfile = createAction(actionTypes.CHANGE_PROFILE);
// export const changeProfile = (profile) => ({
//     type: actionTypes.CHANGE_PROFILE,
//     profile
// });

export const changeToken = createAction(actionTypes.CHANGE_TOKEN);
// export const changeToken = (token) => ({
//     type: actionTypes.CHANGE_TOKEN,
//     token
// });

export const performLogin = (telephone, password) => {
    return (dispatch) => {
        return phoneLogin(telephone, password).then((response) => {
            console.log(response);
            if (response.statusCode != 200) {
                message.error('账号或密码错误');
            }
            else {
                const token = response.data.token;
                dispatch(changeLoginState(true));
                dispatch(changeToken(token));
                dispatch(changeIsVisible(false));

                return getProfile();  
            }
        }).then((res) => {
            console.log(res);
            if (res.statusCode != 200) {
                message.error('用户信息获取失败');
            }
            else {
                let data = res.data;
                dispatch(changeProfile({
                    avatarUrl: data.avatar,
                    usrname: data.trueName,
                    telephone,
                    sex: data.sex,
                }));
            }
        }).catch((e) => {
            message.error('登录失败');
            console.error('phoneLoginError', e);
        })
    };
};

export const performLogout = () => {
    return (dispatch) => {
        return logout().then( (res) => {
            if (res.statusCode == 200) {
                dispatch(changeLoginState(false));
            }
            else {
                throw new Error('statusCode' + res.statusCode);
            }
        }).catch((e) => {
            message.error('退出登录失败');
            console.error('logoutError', e);
        })
    };
};

// export const updateAvatar = (avatarFile) => {
//     return (dispatch) => {
//         sendAvatar(avatarFile).then( (res) => {
//             if (res.statusCode == 200) {
//                 dispatch(changeAvatar(res.data.avatarUrl));
//             }
//             else {
//                 message.error('头像上传失败');
//             }
//         }).catch((e) => console.error('sendAvatarError', e.message))
//     };
// };
