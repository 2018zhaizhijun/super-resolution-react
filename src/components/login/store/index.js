import reducer from './reducer';
import { changeIsVisible, changeProfile, performLogin, performLogout } from './actionCreator';

export {
    reducer,
    changeIsVisible,
    changeProfile,
    performLogin,
    performLogout,
    //updateAvatar,
}