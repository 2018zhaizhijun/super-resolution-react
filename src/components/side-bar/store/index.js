import reducer from './reducer';
import { changeIsVisible as changeSidebarVisible } from './actionCreator';

export {
    reducer,
    changeSidebarVisible,
}