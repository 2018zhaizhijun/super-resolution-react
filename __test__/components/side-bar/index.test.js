import React from 'react';
import SideBar from '@components/side-bar';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '@store';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';

describe('side bar render', () => {

  it('side bar should match snapshot', () => {
    const barWrapper = renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                <SideBar />
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(barWrapper).toMatchSnapshot();
  });

  it('different navigation buttons of sidebar should lead to correct url path', () => {
    const wrapper = mount(
        <Provider store={store}>
            <BrowserRouter>
                <SideBar />
            </BrowserRouter>
        </Provider>
    );

    expect(wrapper.find('#toTransPage').first().prop('to'))
        .toEqual('/main/transform');

    expect(wrapper.find('#toIntroPage').first().prop('to'))
        .toEqual('/main/discribe');

  })

})