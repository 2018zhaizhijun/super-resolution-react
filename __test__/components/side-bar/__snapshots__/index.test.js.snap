// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side bar render side bar should match snapshot 1`] = `
<div
  className="sc-bdvvtL bHgvbk sidebarWrapper"
  style={
    Object {
      "left": "-100%",
    }
  }
>
  <a
    aria-current={null}
    className="sidebar-link"
    href="/main/transform"
    id="toTransPage"
    onClick={[Function]}
  >
    <span>
      超分转换
    </span>
  </a>
  <a
    aria-current={null}
    className="sidebar-link"
    href="/main/discribe"
    id="toIntroPage"
    onClick={[Function]}
  >
    <span>
      功能简介
    </span>
  </a>
</div>
`;
